src/main/scala/com/databricks/spark/sql/perf/mllib/feature/VectorAssembler.scala

```diff
@@ -0,0 +1,49 @@
+package com.databricks.spark.sql.perf.mllib.feature
+
+import org.apache.spark.ml
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.ml.PipelineStage
+import org.apache.spark.sql._
+import com.databricks.spark.sql.perf.mllib.OptionImplicits._
+import com.databricks.spark.sql.perf.mllib.data.DataGenerator
+import com.databricks.spark.sql.perf.mllib.{BenchmarkAlgorithm, MLBenchContext, TestFromTraining}
+
+/** Object for testing VectorAssembler performance */
+object VectorAssembler extends BenchmarkAlgorithm with TestFromTraining {
+
+  val numInputCols = 3
```

Comment:
You could use the `numFeatures` param here, although that might violate convention (typically numFeatures is the number of input features to an estimator).